File: 0234-palindrome-linked-list/0234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def isPalindrome(self, head):
        ptr=head
        str1=""
        while ptr!=None:
            str1+=str(ptr.val)
            ptr=ptr.next
        i=0
        f=len(str1)-1
        while i<=f:
            if str1[i]==str1[f]:
                i+=1
                f-=1
            else:
                return False
        return True

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        HashMap<ListNode, ListNode> map = new HashMap<>();
        ListNode ptr = head, pre = null;

        while(ptr!=null) {
            map.put(ptr,pre);
            pre = ptr;
            ptr = ptr.next;
        }
        ptr = head;

        while(ptr!= pre) {
            if(ptr.val!=pre.val) return false;
            ptr = ptr.next;
            if(ptr == pre) break;
            pre = map.get(pre);
        }
        return true;
    }
}